modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeReplacementsCheck.java

```diff
@@ -0,0 +1,371 @@
+					newContent = _formatParameters(
+						classNames, fileName, from, javaMethodContent,
+						jsonObject, matcher, newContent, to);
+				}
+				else {
+					newContent = StringUtil.replaceFirst(
+						newContent, methodCall,
+						StringUtil.replace(methodCall, from, to));
+				}
+			}
+		}
+
+		if (!content.equals(newContent)) {
+			newContent = _addReplacementDependencies(
+				fileName, jsonObject, newContent);
+		}
+
+		return newContent;
+	}
+
+	private String _formatParameters(
+		String[] classNames, String fileName, String from,
+		String javaMethodContent, JSONObject jsonObject, Matcher matcher,
+		String newContent, String to) {
+
+		String methodCall = JavaSourceUtil.getMethodCall(
+			javaMethodContent, matcher.start());
+
+		List<String> parameterList = JavaSourceUtil.getParameterList(
+			methodCall);
+
+		List<String> parametersTypes = JavaSourceUtil.getParameterList(from);
+
+		if (parameterList.size() != parametersTypes.size()) {
+			return newContent;
+		}
+
+		if (fileName.endsWith(".java") &&
+			!hasParameterTypes(
+				javaMethodContent, javaMethodContent,
+				ArrayUtil.toStringArray(parameterList),
+				ArrayUtil.toStringArray(parametersTypes))) {
+
+			StringBundler sb = new StringBundler(6);
+
+			sb.append("Unable to format ");
+
+			int index = from.indexOf(CharPool.OPEN_PARENTHESIS);
+
+			if (from.contains(StringPool.PERIOD)) {
+				sb.append(from.substring(from.indexOf(CharPool.PERIOD), index));
+			}
+			else {
+				sb.append(from.substring(0, index));
+			}
+
+			sb.append(" method from ");
+
+			if (classNames.length > 0) {
+				sb.append(
+					StringUtil.merge(classNames, StringPool.COMMA_AND_SPACE));
+			}
+			else {
+				sb.append(getVariableName(methodCall));
+			}
+
+			sb.append(". Fill the new parameters manually, see ");
+			sb.append(jsonObject.getString("LPS"));
+
+			addMessage(fileName, sb.toString());
+
+			return newContent;
+		}
+
+		List<String> parametersOrder = JavaSourceUtil.getParameterList(to);
+
+		String newMethodCall = to.substring(
+			0, to.indexOf(CharPool.OPEN_PARENTHESIS) + 1);
+
+		if (!newMethodCall.contains(StringPool.PERIOD)) {
+			newMethodCall = StringBundler.concat(
+				getVariableName(methodCall), CharPool.PERIOD, newMethodCall);
+		}
+
+		newMethodCall = _addOrReplaceParameters(
+			newMethodCall, parameterList, parametersOrder);
+
+		return StringUtil.replaceFirst(newContent, methodCall, newMethodCall);
+	}
+
+	private JSONArray _getReplacementsJSONArray(String fileName)
+		throws Exception {
+
+		Class<?> clazz = getClass();
+
+		ClassLoader classLoader = clazz.getClassLoader();
+
+		InputStream inputStream = classLoader.getResourceAsStream(
+			"dependencies/" + fileName);
+
+		if (inputStream == null) {
+			return new JSONArrayImpl();
+		}
+
+		return new JSONArrayImpl(StringUtil.read(inputStream));
+	}
+
+	private boolean _hasValidClassName(
+		String[] classNames, String content, String methodCall) {
+
+		boolean valid = false;
```

Comment:
@nicolas-sss, can we use `return true` and `return false` and save the declaration of this variable?